app/forms/InflationProofForm.scala

```diff
@@ -17,44 +17,36 @@
 package forms
 
 import com.google.inject.Singleton
-import javax.inject.Inject
-import models.{GmpDate, InflationProof}
+import forms.helper.Mappings
+import models.InflationProof
 import play.api.data.Form
 import play.api.data.Forms._
 import play.api.i18n.{Messages, MessagesImpl}
 import play.api.mvc.MessagesControllerComponents
-@Singleton
-class InflationProofForm  @Inject()(mcc: MessagesControllerComponents) {
-  implicit lazy val messages: Messages = MessagesImpl(mcc.langs.availables.head, mcc.messagesApi)
 
+import java.time.LocalDate
+import javax.inject.Inject
 
-  val YEAR_FIELD_LENGTH: Int = 4
-
-  def dateMustBePresentIfRevaluationWanted(x: InflationProof): Boolean = {
-    x.revaluate match {
-      case Some("Yes") => x.revaluationDate.day.isDefined || x.revaluationDate.month.isDefined || x.revaluationDate.year.isDefined
-      case _ => true
-    }
-  }
+@Singleton
+class InflationProofForm @Inject()(mcc: MessagesControllerComponents) extends Mappings {
+  implicit lazy val messages: Messages = MessagesImpl(mcc.langs.availables.head, mcc.messagesApi)
 
-  val inflationProofForm = Form(
-    mapping(
-      "revaluationDate" -> mapping(
-        "day" -> optional(text),
-        "month" -> optional(text),
-        "year" -> optional(text)
-      )(GmpDate.apply)(GmpDate.unapply)
-        .verifying(messages("gmp.error.date.nonnumber"), x => checkForNumber(x.day) && checkForNumber(x.month) && checkForNumber(x.year))
-        .verifying(messages("gmp.error.day.invalid"), x => checkDayRange(x.day))
-        .verifying(messages("gmp.error.month.invalid"), x => checkMonthRange(x.month))
-        .verifying(messages("gmp.error.year.invalid.format"), x => checkYearLength(x.year))
-      ,
-      "revaluate" -> optional(text).verifying(messages("gmp.error.reason.mandatory"),{_.isDefined})
+  def dateCondition(data: Map[String, String]): Boolean = data.get("revaluate").contains("Yes")
+  def inflationProofForm(minYear: Int, maxYear: Int) = {
+    Form(mapping(
+      "revaluationDate" -> gmpDate(
+        maximumDateInclusive = Some(LocalDate.of(maxYear, 4, 5)),
+        minimumDateInclusive = Some(LocalDate.of(minYear, 4, 6)),
+        "revaluationDate.day",
+        "revaluationDate.month",
+        "revaluationDate.year",
+        "revaluationDate",
+        tooRecentArgs = Seq("5 April " + maxYear.toString),
+        tooFarInPastArgs = Seq("6 April " + minYear.toString),
```

Comment:
this should be 4 April - it's because the GMP start date is 4 April 1978 (what the old constraint checkDateOnOrAfterGMPStart used)